vllm/worker/model_runner.py

```diff
@@ -1551,6 +1551,21 @@ def execute_model(
 
         # Compute the logits in the last pipeline stage.
         if not get_pp_group().is_last_rank:
+            if (self.is_driver_worker
+                    and hidden_or_intermediate_states is not None
+                    and isinstance(hidden_or_intermediate_states,
+                                   IntermediateTensors)
+                    and self.observability_config is not None
+                    and self.observability_config.collect_model_forward_time):
+                model_forward_end.synchronize()
```

Comment:
Just to confirm, does this sync hurt the performance in normal runs? I guess not because you won't have observability config?

Reply:
Yes, this path won't be taken because observability_config.collect_model_forward_time won't be set.